README.md

```diff
@@ -26,70 +24,78 @@ To install Go, visit [this link](https://golang.org/dl/).
 ## Usage
 Before using MSAL Go, you will need to [register your application with the Microsoft identity platform](https://docs.microsoft.com/azure/active-directory/develop/quickstart-v2-register-an-app).
 
-### Public Surface
-
-The Public API of the library can be found in the following directories under `apps`.
+### Acquiring Tokens
 
-```
-apps/ - Contains all our code
-  confidential/ - The confidential application API
-  public/ - The public application API
-  cache/ - The cache interface that can be implemented to provide persistence cache storage of credentials
-```
+Acquiring tokens with MSAL Go follows this general pattern. There might be some slight differences for other token acquisition flows. Here is a basic example:
 
-Acquiring tokens with MSAL Go follows this general three step pattern. There might be some slight differences for other token acquisition flows. Here is a basic example:
+1. Create a client. MSAL separates [public and confidential client applications](https://tools.ietf.org/html/rfc6749#section-2.1), so call `public.New()` or `confidential.New()` to create the appropriate client for your application.
 
-1. MSAL separates [public and confidential client applications](https://tools.ietf.org/html/rfc6749#section-2.1). So, you would create an instance of a `PublicClientApplication` and `ConfidentialClientApplication` and use this throughout the lifetime of your application.
    * Initializing a public client:
 
     ```go
-    publicClientApp, err := public.New("client_id", public.WithAuthority("https://login.microsoft.com/Enter_The_Tenant_Name_Here"))
+    import "github.com/AzureAD/microsoft-authentication-library-for-go/apps/public"
+
+    publicClient, err := public.New("client_id", public.WithAuthority("https://login.microsoftonline.com/your_tenant"))
     ```
 
    * Initializing a confidential client:
 
     ```go
-    // Initializing the client credential
+    import "github.com/AzureAD/microsoft-authentication-library-for-go/apps/confidential"
+
+    // confidential clients have a credential, such as a secret or a certificate
     cred, err := confidential.NewCredFromSecret("client_secret")
     if err != nil {
-        return nil, fmt.Errorf("could not create a cred from a secret: %w", err)
+        // TODO: handle error
     }
-    confidentialClientApp, err := confidential.New("client_id", cred, confidential.WithAuthority("https://login.microsoft.com/Enter_The_Tenant_Name_Here"))
+    confidentialClient, err := confidential.New("https://login.microsoftonline.com/your_tenant", "client_id", cred)
```

Comment:
Out of curiosity, the confidential.New(...) accepts the authority from the first param which is a string, but the publicNew(...) accepts authority from the second param which is an output of WithAuthority helper. Is such an asymmetry indeed the case?

Reply:
Yes. Public client has a reasonable default authority but confidential client doesn't, so only the confidential client constructor requires an authority.

Reply:
Precisely speaking, only confidential client's `AcquireTokenByCredential()` requires a tenanted authority. Auth code flow can still use ".../common" or ".../organizations". That being said, the current API decision is also reasonable.